### DISCUSSION TITLE
Receive a file when sending from a client without multipart.

### DISCUSSION THREAD

**Discussion by Mint-Joy** [kartikk221/hyper-express#241] [Q&A] [answered]
Browser send payload 

------WebKitFormBoundaryYO9WpPGAq2TVdb7i
Content-Disposition: form-data; name="1691558702172454724.jpg"; filename="1691558702172454724.jpg"
Content-Type: image/jpeg

------WebKitFormBoundaryYO9WpPGAq2TVdb7i--

in handler in req.urlencoded() contains all data, multipart is empty.
Do you need to parse it yourself or are there some mechanisms inside hyper-express that allow you to get processed data?

**Comment by kartikk221** [comment, marked as answer]
Hi, you should use multipart because multipart allows for the file to be gradually streamed.

Alternatively, what you can do is pipe the `Request` to one of your own form handlers as the `Request` for `hyper-express` is a `ReadableStream` which can be used to consume raw body data.

I will say your payload does look like a multi-part payload to me, so why don't you try following this guide to implement a potential solution https://github.com/kartikk221/hyper-express/blob/master/docs/MultipartField.md

**Comment by Mint-Joy** [comment]
I did it as in your example
```
async function uploadsIndex(req, res, next) {
    try {
        await req.multipart(async (field) => {
            let aField = await field;
            console.log('field', field)
            console.log('aField', aField)
        })
    } catch (error) {
        // The multipart parser may throw a string constant as an error
        // Be sure to handle these as stated in the documentation
        if (typeof error === "FILES_LIMIT_REACHED") {
            return res.status(403).send("You sent too many files! Try again.")
        } else {
            return res.status(500).send("Oops! An uncaught error occured on our end.")
        }
    }
}
```
console output

```
field undefined
aField undefined
```
if I send from a client
`Content-Type: multipart/form-data`
The parsing works, but there are several clients and not all can create the required headers. The team that writes the client says that everything works fine on **Adonis.js**

I temporarily solved it using **formidable**

```
    const form = formidable({})
    let fields
    let files
    try {
        ;[fields, files] = await form.parse(req)
    } catch (err) {
        res.status(400).json("Data error")
        return
    }
```    
console output

```
{
        "fields": {},
        "files": {
            "1691558702172454724.jpg": [
                {
                    "size": 14710,
                    "filepath": "C:\\Users\\fast\\AppData\\Local\\Temp\\dd1bd1f414bec4c6d39de1e01",
                    "newFilename": "dd1bd1f414bec4c6d39de1e01",
                    "mimetype": "image/jpeg",
                    "mtime": "2024-03-26T20:05:21.231Z",
                    "originalFilename": "1691558702172454724.jpg"
                }
            ]
        }
    }
```
But I think that the framework should solve such issues effortlessly. This is a completely standard situation.
This doesn’t even need to be written in a cookbook; such problems simply shouldn’t exist. Hyper Express is a wonderful framework, and when it gets rid of such little things it will become completely production ready.
Good luck to all.

**Reply by kartikk221** [reply]
- I am bit confused as to why you are awaiting the `field` in the multipart handler in the first example.
- HyperExpress uses `busboy` under the hood which is the exact same parser that many body parsing middlewares for Express.js use hence supporting MultiPart for uploads is implemented for compatibility as it is the most common use case.

I am not sure why the `field` argument is `undefined` in the `multipart` handler as that should never be the case and can be something we explore as a bug though I would need a better reproduction.
